### PR TITLE
🐛 Fixed workspace subscription used by database ip whitelist component

### PR DESCRIPTION
**Pull request by yjmrobert**
# Pull Request

## Description

Removed redundant usings and refactored database IP whitelist logic. Added new SpecFlow tests to validate the subscription ID fetching process in the database IP whitelist component, ensuring proper workspace subscription ID use.

## Changes

- Removed redundant usings
- Switched subscription to workspace's subscription

## Testing

- Added gherkin test to verify future issues

## Checklist

- [x] Code follows dotnet coding standards
- [x] Tests added/updated to cover changes

### CODE REVIEW

File: Portal/test/Datahub.SpecflowTests/Datahub.SpecflowTests.csproj [hunk 1]
```diff
@@ -39,4 +39,8 @@
     <ItemGroup>
         <Content Include="Features\**\*.feature" />
     </ItemGroup>
+
+    <ItemGroup>
+      <Compile Remove="Hooks\IpAddressWhitelistStepsHook.cs" />
+    </ItemGroup>
```
Comment (yjmrobert) [inline Portal/test/Datahub.SpecflowTests/Datahub.SpecflowTests.csproj]:
```suggestion
```

File: Portal/test/Datahub.SpecflowTests/Features/IPAddressWhitelist.feature [hunk 2]
```diff
@@ -1,5 +1,11 @@
+@IWebHostEnvironment
```
Comment (yjmrobert) [inline Portal/test/Datahub.SpecflowTests/Features/IPAddressWhitelist.feature]:
```suggestion
```

**Review by vb-ssc — APPROVED** [review]
Looks good

**Review by Sean-Stilwell — APPROVED** [review]
Tested locally, LGTM!

**Review by davidreneuw — APPROVED** [review]

**Review by ErikApption — APPROVED** [review]

**Pull request merged**